compiler/parser/support.go

```diff
@@ -33,20 +57,54 @@ func makeArgMap(args interface{}) (interface{}, error) {
 	return m, nil
 }
 
-func makeTemplateExprChain(in interface{}) interface{} {
-	rest := in.([]interface{})
-	ret := rest[0]
+func makeTemplateExprChain(in any) any {
+	rest := in.([]any)
+	ret := rest[0].(ast.Expr)
 	for _, part := range rest[1:] {
-		ret = map[string]interface{}{
-			"kind": "BinaryExpr",
-			"op":   "+",
-			"lhs":  ret,
-			"rhs":  part,
+		ret = &ast.BinaryExpr{
+			Kind: "BinaryExpr",
+			Op:   "+",
+			LHS:  ret,
+			RHS:  part.(ast.Expr),
 		}
 	}
 	return ret
 }
 
+func newCall(c *current, name, args, where any) ast.Expr {
+	call := &ast.Call{
+		Kind:    "Call",
+		Name:    name.(string),
+		NamePos: c.pos.offset,
+		Args:    sliceOf[ast.Expr](args),
+		Rparen:  lastPos(c, ")"),
+	}
+	if where != nil {
+		call.Where = where.(ast.Expr)
+	}
+	return call
+}
+
+func lastPos(c *current, s string) int {
+	i := bytes.LastIndex(c.text, []byte(s))
+	if i == -1 {
+		panic(fmt.Sprintf("system error: character %s not found in %s", s, string(c.text)))
+	}
+	return c.pos.offset + i
+}
+
+func firstPos(c *current, s string) int {
+	i := bytes.Index(c.text, []byte(s))
+	if i == -1 {
+		panic(fmt.Sprintf("system error: character %s not found in %s", s, string(c.text)))
+	}
+	return c.pos.offset
```

Comment:
Shoudl this be `c.pos.offset + i`?

Reply:
Whoops!

Reply:
Oh this isn't even being used removed...